lib/internal/util/diff.js

```diff
@@ -0,0 +1,204 @@
+  let message = printSimpleMyersDiff(diff);
+
+  if (skipped) {
+    message += '...';
+  }
+
+  return { message, header, skipped };
+}
+
+function getStackedDiff(actual, expected) {
+  const isStringComparison = typeof actual === 'string' && typeof expected === 'string';
+
+  let message = `\n${colors.green}+${colors.white} ${actual}\n${colors.red}- ${colors.white}${expected}`;
+  const stringsLen = actual.length + expected.length;
+  const maxTerminalLength = process.stderr.isTTY ? process.stderr.columns : 80;
+  const showIndicator = isStringComparison && (stringsLen <= maxTerminalLength);
+
+  if (showIndicator) {
+    let indicatorIdx = -1;
+
+    for (let i = 0; i < actual.length; i++) {
+      if (actual[i] !== expected[i]) {
+        // Skip the indicator for the first 2 characters because the diff is immediately apparent
+        // It is 3 instead of 2 to account for the quotes
+        if (i >= 3) {
+          indicatorIdx = i;
+        }
+        break;
+      }
+    }
+
+    if (indicatorIdx !== -1) {
+      message += `\n${StringPrototypeRepeat(' ', indicatorIdx + 2)}^`;
+    }
+  }
+
+  return { message };
+}
+
+function getSimpleDiff(originalActual, actual, originalExpected, expected) {
+  const kMaxShortStringLength = 12;
+
+  let stringsLen = actual.length + expected.length;
+  // Accounting for the quotes wrapping strings
+  if (typeof originalActual === 'string') {
+    stringsLen -= 2;
+  }
+  if (typeof originalExpected === 'string') {
+    stringsLen -= 2;
+  }
+  if (stringsLen <= kMaxShortStringLength && (originalActual !== 0 || originalExpected !== 0)) {
+    return { message: `${actual} !== ${expected}`, header: '' };
+  }
+
+  const isStringComparison = typeof originalActual === 'string' && typeof originalExpected === 'string';
+
+  if (isStringComparison && colors.hasColors) {
+    return getColoredMyersDiff(actual, expected);
+  }
+
+  return getStackedDiff(actual, expected);
+}
+
+function isSimpleDiff(actual, inspectedActual, expected, inspectedExpected) {
+  if (inspectedActual.length > 1 || inspectedExpected.length > 1) {
+    return false;
+  }
+
+  return typeof actual !== 'object' || actual === null || typeof expected !== 'object' || expected === null;
+}
+
+function getErrorMessage(operator, message) {
+  if (!operator && !message) {
+    return '';
+  }
+  return message || kReadableOperator[operator];
+}
+
+/**
+ * Generate a difference report between two values
+ * @param {any} actual - The first value to compare
+ * @param {any} expected - The second value to compare
+ * @returns {string|null} - Formatted diff string or null if values are equal
+ */
+// The following parameters are not documented because they are only used internally
+// @param {string} [operator] - The operator used in the assertion
+// @param {string} [customMessage] - Custom message to display in the diff, still just in the assertion
+function diff(actual, expected, operator, customMessage) {
```

Comment:
I am confused about this. The parameters do exist in the exposed form, if I am not mistaken.

Can we please just expose a wrapper method that does not have these arguments? If something is internal, it should really stay as such. Everything that Node.js exposes is public API, no matter if it meant for internal usage or not. Be certain that it will be used :)